crates/types/src/simple_certificate.rs

```diff
@@ -0,0 +1,90 @@
+#![allow(dead_code)]
+#![allow(clippy::missing_docs_in_private_items)]
+#![allow(missing_docs)]
+
+use std::marker::PhantomData;
+
+use commit::Commitment;
+use ethereum_types::U256;
+
+use crate::{
+    simple_vote::{Voteable, YesData},
+    traits::{
+        election::Membership, node_implementation::NodeType, signature_key::SignatureKey,
+        state::ConsensusTime,
+    },
+    vote2::Certificate2,
+};
+
+/// A certificate which can be created by aggregating many simple votes on the commitment.
+#[derive(Eq, Hash, PartialEq, Debug, Clone)]
+pub struct SimpleCertificate<TYPES: NodeType, VOTEABLE: Voteable, MEMBERSHIP: Membership<TYPES>> {
```

Comment:
I'm branching off this branch for my work.  Is this supposed to represent all the SimpleCertificates, or just the QuorumCertificate type?

Reply:
It should be eventually all simple certificates.  The exception right now is that the threshold part isn't properly implemented for certificates that use `failure_threshold()` of the `MEMBERSHIP`

Reply:
If that is the case, then I don't think it makes sense to have leaf_commitment be a field of the certificate?

Reply:
It's just poorly named, it is always the thing which is voted on

Reply:
Could we make the SimpleCertificate generic over the threshold?  That way we can use the same SimpleCertificate for all certs?

Reply:
Ideally we can.  I didn't think of a good way to do this which works with the types, and the `VoteAccumulator`